### PR TITLE
Creating input

### PR DESCRIPTION
**Pull request by thomashenson**
Please do the reviews.

### CODE REVIEW

File: DataGen.py [hunk 1]
```diff
@@ -128,10 +133,22 @@ def main(argv):
                 # add this response to output data frame
                 output_df = output_df.append(response_df, ignore_index=True)
 
+                if len(output_df.index) > 350:
+                    output_df['land_or_marine'][enterprise_index] = "M"
+
```
Comment (Piwington) [inline DataGen.py]: This isn't great. Have spoken about other suggestions

File: DataGen.py [hunk 2]
```diff
@@ -128,10 +133,22 @@ def main(argv):
...
 
+    # split into current and previous period files
+    # unless split=False
+    if split is True:
+        previous_df = output_df[output_df['period'] == 201812]
+        current_df = output_df[output_df['period'] == 201903]
+
+        previous_df.to_json("fixtures/previous_out.json", orient="records")
+        current_df.to_json("fixtures/current_out.json", orient="records")
+
```
Comment (Piwington) [inline DataGen.py]: May wish to do a pass in for these file names?

File: DataGen.py (hunk 1 above)
Comment (KrisRogos) [inline DataGen.py]: Is this necessary? If so could the threshold (350), column name (land_or_marine, enterprise_index) and value (M) be parameterised via the JSON config?

File: DataGen.py [hunk 3]
```diff
@@ -128,10 +133,22 @@ def main(argv):
...
     # save to csv
     output_df = output_df.ix[:, column_order]
     output_df.to_csv(output_file_path, header=False, index=False)
 
+    # split into current and previous period files
+    # unless split=False
+    if split is True:
+        previous_df = output_df[output_df['period'] == 201812]
+        current_df = output_df[output_df['period'] == 201903]
```
Comment (KrisRogos) [inline DataGen.py]: JSON config allows to specify any period, what happens if I enable split while my file has periods 201903 and 201906, or 202003 and 202006?

File: fixtures/sample_survey.json [hunk 4]
```diff
@@ -1,71 +1,125 @@
...
+        "Q604_bituminous_gravel",
+        "Q605_concreting_gravel",
+        "Q606_other_gravel",
+        "Q607_constructional_fill",
+        "Q608_total",
+        "land_or_marine",
+        "manifest_rowcount",
+        "manifest_runid",
+        "manifest_system",
+        "manifest_timestamp"
```
Comment (KrisRogos) [inline fixtures/sample_survey.json]: The sample_survey.json should be only a very basic demo one, not contain details of our actual survey structure. The 'real' configs should be kept on the internal network share. This applies to the rest of this document too.

File: DataGen.py (hunk 2 above)
Comment (thomashenson) [reply DataGen.py]: Could make this so for bricks and blocks. Will implement this.

File: DataGen.py (hunk 1 above)
Comment (thomashenson) [reply DataGen.py]: Has been removed.

File: DataGen.py (hunk 3 above)
Comment (thomashenson) [reply DataGen.py]: Now resolved.

File: fixtures/sample_survey.json (hunk 4 above)
Comment (thomashenson) [reply fixtures/sample_survey.json]: Resolved.

File: DataGen.py (hunk 1 above)
Comment (thomashenson) [reply DataGen.py]: Removed.

**Comment by Piwington** [comment]
While uploading the JSON config is probably fine, I would suggest not having the output files on git.

**Comment by thomashenson** [comment]
> While uploading the JSON config is probably fine, I would suggest not having the output files on git.

It could be removed, but it's all dummy data, so not necessarily harmful. The out.csv resembles the other outputs and that is already on git, hence why I didn't think it an issue. Unless of course, it's the var names you're referring to? Either way, I don't mind removing them.

File: DataGen.py (hunk 2 above)
Comment (thomashenson) [reply DataGen.py]: Resolved.

**Comment by pwetherill** [comment]
@thomashenson 
sample_survey.json has a typo for one of the data_frame_columns!
Should read "enterprise_name" with 1 't'

File: DataGen.py [hunk 5]
```diff
@@ -133,21 +146,29 @@ def main(argv):
...
-
     # save to csv
     output_df = output_df.ix[:, column_order]
-    output_df.to_csv(output_file_path, header=False, index=False)
+    output_df.to_csv(output_file_path, header=True, index=False)
+
+    # display last ruref
+    print("------------------------------------------------------")
+    print("Starting ruref for land/marine: " + str(ruref))
+    print("------------------------------------------------------")
```
Comment (KrisRogos) [inline DataGen.py]: What if this isn't s&g?

File: DataGen.py [hunk 6]
```diff
@@ -10,28 +10,37 @@
 
 def main(argv):
 
-    survey_config_file_path = 'fixtures/sample_survey.json'
+    survey_config_file_path = 'fixtures/SynthData/SandAndGravelLand.json'
```
Comment (KrisRogos) [inline DataGen.py]: Do we want this to be the default?

**Pull request merged**